app/controllers/topics_controller.rb

```diff
@@ -897,10 +897,21 @@ def reset_new
       if params[:include_subcategories] == 'true'
         category_ids = category_ids.concat(Category.where(parent_category_id: params[:category_id]).pluck(:id))
       end
-      DismissTopics.new(current_user, Topic.where(category_id: category_ids)).perform!
+
+      topic_scope =
+        if params[:tag_id]
+          Topic.where(category_id: category_ids).joins(:tags).where(tags: { name: params[:tag_id] })
```

Comment:
this is a bit confusing to me ... can category_ids be `[]` when you are looking at the `/new` with no filters?

Reply:
If a category is not selected it is going to else branch (which will be solved with my next PR - and that logic will be a bit simplified once we have everything using DismissTopicService)

```ruby
  def reset_new
    if params[:category_id].present?
      category_ids = [params[:category_id]]
      if params[:include_subcategories] == 'true'
        category_ids = category_ids.concat(Category.where(parent_category_id: params[:category_id]).pluck(:id))
      end

      topic_scope =
        if params[:tag_id]
          Topic.where(category_id: category_ids).joins(:tags).where(tags: { name: params[:tag_id] })
        else
          Topic.where(category_id: category_ids)
        end

      DismissTopics.new(current_user, topic_scope).perform!
      category_ids.each do |category_id|
        TopicTrackingState.publish_dismiss_new(current_user.id, category_id: category_id, tag_id: params[:tag_id])
      end
    elsif params[:tag_id].present?
      DismissTopics.new(current_user, Topic.joins(:tags).where(tags: { name: params[:tag_id] })).perform!
      TopicTrackingState.publish_dismiss_new(current_user.id, tag_id: params[:tag_id])
    else
      if params[:tracked].to_s == "true"
        topics = TopicQuery.tracked_filter(TopicQuery.new(current_user).new_results, current_user.id)
        topic_users = topics.map { |topic| { topic_id: topic.id, user_id: current_user.id, last_read_post_number: 0 } }
        TopicUser.insert_all(topic_users) if !topic_users.empty?
      else
        current_user.user_stat.update_column(:new_since, Time.zone.now)
        TopicTrackingState.publish_dismiss_new(current_user.id)
      end
    end
 ```